R/MsBackendDataFrame-functions.R

```diff
@@ -92,9 +101,53 @@ NULL
     collisionEnergy = "numeric"
 )
 
+#' accessor methods for spectraData columns.
+#'
+#' @noRd
+.SPECTRA_DATA_COLUMN_METHODS <- c(
+    msLevel = "msLevel",
+    rt = "rtime",
+    acquisitionNum = "acquisitionNum",
+    scanIndex = "scanIndex",
+    mz = "mz",
+    intensity = "intensity",
+    fromFile = "fromFile",
+    centroided = "centroided",
+    smoothed = "smoothed",
+    polarity = "polarity",
+    precScanNum = "precScanNum",
+    precursorMz = "precursorMz",
+    precursorIntensity = "precursorIntensity",
+    precursorCharge = "precursorCharge",
+    collisionEnergy = "collisionEnergy"
+)
+
 #' @rdname MsBackend
 #'
-#' @export MsBackendMemory
-MsBackendMemory <- function() {
-    new("MsBackendMemory")
+#' @export MsBackendDataFrame
+MsBackendDataFrame <- function() {
+    new("MsBackendDataFrame")
+}
+
+#' Helper function to extract a certain column from the spectraData data frame.
+#' If the data frame has no such column it will use the accessor method to
+#' retrieve the corresponding data.
+#'
+#' @param x object with a `@spectraData` slot containing a `DataFrame`.
+#'
+#' @param column `character(1)` with the column name.
+#'
+#' @author Johannes Rainer
+#'
+#' @noRd
+.get_spectra_data_column <- function(x, column) {
```

Comment:
I don't understand the rational for this one. In the `MsBackendDataFrame`, is all the data not supposed to be stored in `x@spectraData`? Why would there be two ways to access it, using `spectraData` and the accessor? Would the accessor not itself retrieve a `spectraData` column?

Reply:
I don't think it makes sense to keep all data (all mandatory columns) in memory in the `@spectraData` slot. I would keep that flexible. Also have a look at the accessor methods. They will create the expected data and initialize it with `NA` if the column is not in the `@spectraData` slot. Why should we keep a `DataFrame` full of `NA` values?